CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -577,15 +578,42 @@ CF_EXPORT CFURLRef CFCopyHomeDirectoryURLForUser(CFStringRef uName) {
         return result;
     }
 #elif TARGET_OS_WIN32
-    // This code can only get the directory for the current user
-    CFStringRef userName = uName ? CFCopyUserName() : NULL;
-    if (uName && !CFEqual(uName, userName)) {
-        CFLog(kCFLogLevelError, CFSTR("CFCopyHomeDirectoryURLForUser(): Unable to get home directory for other user"));
-        if (userName) CFRelease(userName);
-        return NULL;
-    }
-    if (userName) CFRelease(userName);
-    return CFCopyHomeDirectoryURL();
+    if (uName == NULL)
+      return CFCopyHomeDirectoryURL();
+
+    CFStringRef home = NULL;
+    LPUSER_INFO_1 pBuffer = NULL;
+    DWORD dwEntriesRead = 0;
+    DWORD dwEntries = 0;
+    DWORD dwToken = 0;
+    NTSTATUS nStatus;
+    do {
+      nStatus = NetUserEnum(NULL, 1, FILTER_NORMAL_ACCOUNT, (LPBYTE*)&pBuffer,
+                            MAX_PREFERRED_LENGTH, &dwEntriesRead, &dwEntries,
+                            &dwToken);
+      if (nStatus == NERR_Success || nStatus == ERROR_MORE_DATA) {
+        for (unsigned uiEntry = 0; !home && uiEntry < dwEntriesRead; ++uiEntry) {
+          CFStringRef name =
+              CFStringCreateWithCStringNoCopy(kCFAllocatorSystemDefault,
+                                              pBuffer[uiEntry].usri1_name,
+                                              kCFStringEncodingUTF16,
+                                              kCFAllocatorNull);
+
+          if (CFEqual(uName, name))
```

Comment:
Code style nit, but we require `{ }` if not a 1-line `if`. Same applies on line 582.